web/src/stream_list.ts

```diff
@@ -981,6 +981,11 @@ function on_sidebar_channel_click(
         e.stopPropagation();
     }
 
+    const section_for_stream = stream_list_sort.current_section_id_for_stream(stream_id);
```

Comment:
This probably deserves a comment of the form:

> In the event that user clicks on the channel when in a collapsed folder, which is only possible if the user was already viewing that channel, we uncollapse the section because ...